README.org

```diff
@@ -196,29 +196,37 @@ Several options are available through Emacs' Customize interface under
 
   Default: =t=
 
-- =winum-assign-func=
+- =winum-assign-functions=
 
-  Function called for each window by =winum-mode=. This is called before
-  automatic assignment begins. The function should return a number to have it
-  assigned to the current-window, =nil= otherwise.
+  List of functions called for each window by `winum-mode'.
 
-  This function along with `winum-auto-assign-0-to-minibuffer' are the only ways
-  to have 0 assigned to a window.
+  These functions allow for deterministic assignment of numbers to windows. Each
+  function is called for every window. A function should return the number to be
+  assigned to a window or nil. The /first/ function to output a number for
+  a given window will determine this window's number.
+
+  If the list is empty or if every function returns nil for a given window winum
+  will proceed to automatic number assignment.
+
+  Since this list is meant to allow custom window assignment for /mutiple/
+  packages at once it should never be directly set, only added to and removed
+  from.
+
+  These functions, along with ~winum-auto-assign-0-to-minibuffer~, are the only
+  way to have 0 assigned to a window.
 
   Example: always assign *Calculator* the number 9 and *NeoTree* the number 0:
 
-#+BEGIN_SRC emacs-lisp
-  (defun my-winum-assign-func ()
-    (cond
-     ((equal (buffer-name) "*Calculator*")
-      9)
-     ((string-match-p (buffer-name) ".*\\*NeoTree\\*.*")
-      0)
-     (t
-      nil)))
-
-  (setq winum-assign-func 'my-winum-assign-func)
-#+END_SRC
+  #+BEGIN_SRC emacs-lisp
```

Comment:
I think it's best if one of your two functions did like the example that you replaced, manage several number/buffer in the same function, so that we show the two ways of doing it in one example :smile_cat: